express-delete/server.ts

```diff
@@ -0,0 +1,53 @@
+import express from 'express';
+
+type Grade = {
+  id: number;
+  name: string;
+  course: string;
+  score: number;
+};
+
+const grades: Record<number, Grade> = {
+  12: {
+    id: 12,
+    name: 'Tim Berners-Lee',
+    course: 'HTML',
+    score: 95,
+  },
+  47: {
+    id: 47,
+    name: 'Brendan Eich',
+    course: 'JavaScript',
+    score: 100,
+  },
+  273: {
+    id: 273,
+    name: 'Håkon Wium Lie',
+    course: 'CSS',
+    score: 92,
+  },
+};
+
+const app = express();
+
+app.get('/api/grades/', (req, res) => {
+  const gradesArray: Grade[] = [];
+  for (const g in grades) {
+    gradesArray.push(grades[g]);
+  }
+  res.json(gradesArray);
+});
+
+app.delete('/api/grades/:id', (req, res) => {
+  const gradeID = req.params.id;
+  if (!(gradeID in grades)) {
```

Comment:
This might fail because `gradeID` is a string though the keys are numbers. The behavior of JS is a complex in this case, though, so I suspect it will work. Did you test it?

The safest thing to do is to put the `+` on line 42 so that `gradeID` is always a number.